src/cpp/src/visual_language/vision_encoder.hpp

```diff
@@ -86,15 +87,17 @@ class VisionEncoder {
     /// instead of the config obtained in constructors.
     /// @return Resulting embeddings for the resized source image and
     /// its slices.
-    virtual EncodedImage encode(const ov::Tensor& image, const ov::AnyMap& config_map = {}) = 0;
+    EncodedImage encode(const ov::Tensor& image, const ov::AnyMap& config_map = {});
+    virtual EncodedImage encode(ov::InferRequest& encoder, const ov::Tensor& image, const ov::AnyMap& config_map = {}) = 0;
```

Comment:
should such method be moved to protected section? as it become an implementation detail